### ISSUE TITLE
Question about trying to bind IPv4 & IPv6 on one node/worker

### ISSUE THREAD

**Issue by tcrowe** [haraka/Haraka#2660]
### system info

Haraka | Haraka.js — Version: 2.8.24
 --- | :---
Node | v11.15.0
OS | Linux localhost 4.9.0-8-amd64 #1 SMP Debian 4.9.144-3 (2019-02-02) x86_64 GNU/Linux
openssl | OpenSSL 1.1.0j  20 Nov 2018

### Expected behavior

Given `smtp.ini` `listen=` directive:

```ini
listen=0.0.0.0:25,0.0.0.0:465,0.0.0.0:587,[::0]:25,[::0]:465,[::0]:587
```

Bind interface-port combinations (view with `netstat -luntp`):

```
0.0.0.0:25
0.0.0.0:465
0.0.0.0:587
[::0]:25
[::0]:465
[::0]:587
```

### Observed behavior

1. `0.0.0.0:25` binds (along with the other ipv4 0.0.0.0)
2. Attempt to bind `[::0]:25`
3. Node error `EADDRINUSE`

### Steps to reproduce

Use the `listen=` directive provided above.

---

Thank you for taking a look, good Harakians! 🤗

**Comment by msimerson** [comment]
The first two lines (comments) in smtp.ini:

```ini
; address to listen on (default: all IPv6 and IPv4 addresses, port 25)
; use "[::0]:25" to listen on IPv6 and IPv4 (not all OSes)
```

Therefore, I use this line to accomplish what you wish (bind to IPv4 and IPv6 ports 25, 465, 587.

```ini
listen=[::0]:25,[::0]:465,[::0]:587
```

**Comment by tcrowe** [comment]
Thank you, @msimerson. 👍 I did, indeed, gloss over that!

---

For others searching out for a similar issue, I will describe what I got through trial and error. I wanted to get an understanding of the server and build evidence that it is functioning correctly now.

Using this setting in `smtp.ini`

```ini
listen=[::0]:25,[::0]:465,[::0]:587
```

I expected only IPv6 to bind. To confirm I use `netstat -luntp` as root.

The results *appear* to confirm my expectation:

```
tcp6       0      0 :::587                  :::*                    LISTEN      16552/node
tcp6       0      0 :::465                  :::*                    LISTEN      16552/node
tcp6       0      0 :::25                   :::*                    LISTEN      16552/node
```

Appearances can be deceiving! 🧐 Trusting that our good, @msimerson is correct, I test if my system also responds on the IPv4 address.

Force IPv4 connection:

`socat -4 - tcp:127.0.0.1:25` ✅server greeting
`socat -4 - tcp:127.0.0.1:465` 🤷🏻‍♂️❓
`socat -4 - tcp:127.0.0.1:587` ✅server greeting

Force IPv6 connection:
`socat -6 - "tcp:[::0]:25"` ✅server greeting
`socat -6 - "tcp:[::0]:465"` ❓
`socat -6 - "tcp:[::0]:587"` ✅server greeting

I'm not sure why but port `465` does not respond to poking it. It's TLS after all so lets try that.

`./test-465.js`
```js
const socket = require("tls").connect({ port: 465, host: "127.0.0.1", rejectUnauthorized: false }, function() {
  console.log("connected");
  setTimeout(() => socket.write("---- POKE! show me in the logs! 😛 ----"), 1000);
  setTimeout(() => socket.end(), 2000);
});
```

`node test-465.js`  ✅server greeting

---

Take care, Harakians. 👋
